File: app/src/main/java/org/xdty/callerinfo/contract/MainBottomContact.java
package org.xdty.callerinfo.contract;

import org.xdty.callerinfo.model.db.Caller;
import org.xdty.callerinfo.model.db.InCall;

public interface MainBottomContact {
    interface View extends BaseView<MainContract.Presenter> {

        void init(InCall inCall, Caller caller);

        void updateMark(int viewId, Caller caller);

        void updateMarkName(String name);
    }

    interface Presenter extends BasePresenter {
        void bindData(InCall inCall);

        boolean canMark();

        void markClicked(int viewId);

        void markCustom(String text);
    }
}

File: app/src/main/java/org/xdty/callerinfo/contract/MainBottomContact.kt
package org.xdty.callerinfo.contract

import org.xdty.callerinfo.model.db.Caller
import org.xdty.callerinfo.model.db.InCall

interface MainBottomContact {
    interface View : BaseView<Presenter> {
        fun init(inCall: InCall, caller: Caller)
        fun updateMark(viewId: Int, caller: Caller)
        fun updateMarkName(name: String?)
    }

    interface Presenter : BasePresenter {
        fun bindData(inCall: InCall?)
        fun canMark(): Boolean
        fun markClicked(viewId: Int)
        fun markCustom(text: String?)
    }
}